src/grain/createGrainAllocation.test.js

```diff
@@ -0,0 +1,356 @@
+        receipts,
+      };
+    };
+
+    it("handles an interval with even cred distribution", () => {
+      const result = createGrainAllocation(
+        immediateStrategy,
+        BUDGET,
+        [evenInterval],
+        new Map()
+      );
+      // $ExpectFlowError
+      const HALF = ONE / 2n;
+      const expectedReceipts = [
+        {address: foo, amount: HALF},
+        {address: bar, amount: HALF},
+      ];
+      const expectedAllocation = createImmediateAllocation(expectedReceipts);
+      expectAllocationsEqual(result, expectedAllocation);
+    });
+    it("handles an interval with un-even cred distribution", () => {
+      const result = createGrainAllocation(
+        immediateStrategy,
+        BUDGET,
+        [unevenInterval],
+        new Map()
+      );
+      // $ExpectFlowError
+      const ONE_TENTH = ONE / 10n;
+      const NINE_TENTHS = ONE - ONE_TENTH;
+      const expectedReceipts = [
+        {address: foo, amount: NINE_TENTHS},
+        {address: bar, amount: ONE_TENTH},
+      ];
+      const expectedAllocation = createImmediateAllocation(expectedReceipts);
+      expectAllocationsEqual(result, expectedAllocation);
+    });
+    it("handles an interval with one cred recipient", () => {
+      const result = createGrainAllocation(
+        immediateStrategy,
+        BUDGET,
+        [singlePersonInterval],
+        new Map()
+      );
+      const expectedReceipts = [{address: bar, amount: BUDGET}];
+      const expectedAllocation = createImmediateAllocation(expectedReceipts);
+      expectAllocationsEqual(result, expectedAllocation);
+    });
+  });
+
+  describe("balancedAllocation", () => {
+    const BUDGET = fromApproximateFloat(14);
+
+    const createBalancedAllocation = (
+      receipts: $ReadOnlyArray<GrainReceipt>,
+      budget: Grain
+    ) => {
+      return {
+        version: GRAIN_ALLOCATION_VERSION_1,
+        strategy: balancedStrategy,
+        budget,
+        receipts,
+      };
+    };
+
+    it("should only pay Foo if Foo is sufficiently underpaid", () => {
+      const lifetimeEarnings = new Map([
+        [foo, ZERO],
+        [bar, fromApproximateFloat(99)],
+      ]);
+      const expectedReceipts = [
+        {address: foo, amount: fromApproximateFloat(14)},
+      ];
+      const expectedAllocation = createBalancedAllocation(
+        expectedReceipts,
+        BUDGET
+      );
+      const actual = createGrainAllocation(
+        balancedStrategy,
+        BUDGET,
+        credHistory,
+        lifetimeEarnings
+      );
+      expectAllocationsEqual(expectedAllocation, actual);
+    });
+    it("should divide according to cred if everyone is already balanced paid", () => {
+      const lifetimeEarnings = new Map([
+        [foo, fromApproximateFloat(5)],
+        [bar, fromApproximateFloat(2)],
+      ]);
+
+      const expectedReceipts = [
+        {address: foo, amount: fromApproximateFloat(10)},
+        {address: bar, amount: fromApproximateFloat(4)},
+      ];
```

Comment:
May be worth commenting why we're expecting these values. So readers don't need to look up as many variables. Along the lines of:
The total cred is `foo: 10, bar: 4`, past distributions are exactly half of this and we're distributing 14g. So the expected receipts should match 1:1 with their cred.

Reply:
done